File: dealers/src/main/java/io/mb/dealers/configuration/OpenApiConfiguration.java
package io.mb.dealers.configuration;

import io.mb.dealers.configuration.properties.BuildInfoProperties;
import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Info;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
@RequiredArgsConstructor
public class OpenApiConfiguration {

    private final BuildInfoProperties buildInfoProperties;

    @Bean
    public OpenAPI openAPI() {
        return new OpenAPI().info(
                new Info().version(buildInfoProperties.version())
                        .title(buildInfoProperties.name())
                        .description(buildInfoProperties.description())
        );
    }
}

File: vehicles/src/main/kotlin/io/mb/vehicles/configuration/OpenApiConfiguration.kt
package io.mb.vehicles.configuration

import io.mb.vehicles.configuration.properties.BuildInfoProperties
import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenApiConfiguration(private val buildInfoProperties: BuildInfoProperties) {

    @Bean
    fun openAPI(): OpenAPI = with(buildInfoProperties) {
        OpenAPI().info(Info().version(version).title(name).description(description))
    }
}
